File: LeetCode/LeetCode Practice Problems/242- Valid Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        int a[26] = {0};
        
        for(int i=0;i<s.length();i++) a[s[i] - 'a']++;
        for(int i=0;i<t.length();i++) a[t[i] - 'a']--;
        
        for(int i=0;i<26;i++) 
            if(a[i] != 0) 
                return false;
        
        
        return true;
        
    }
};

File: LeetCode/LeetCode Practice Problems/242-Valid Anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int freq[] = new int[26];
        
        for(char c: s.toCharArray()) {
            freq[c - 'a']++;
        }
        for(char c: t.toCharArray()) {
            freq[c - 'a']--;
        }
        
        for(int i=0;i<26;i++) {
            if(freq[i] != 0) 
                return false;
        }
        
        return true;
    }
}
